src/backend/BSDispatcher/Constraints.pm

```diff
@@ -111,6 +111,7 @@ sub oracle {
     my $memory = ($worker->{'hardware'}->{'memory'} || 0);
     my $swap = ($worker->{'hardware'}->{'swap'} || 0);
     return 0 if $constraints->{'hardware'}->{'memory'} && getmbsize($constraints->{'hardware'}->{'memory'}) > ( $memory + $swap );
+    return 0 if $constraints->{'hardware'}->{'memory'} && getmbsize($constraints->{'hardware'}->{'memoryperjob'} * ($worker->{'hardware'}->{'jobs'} || 0)) > ( $memory + $swap );
```

Comment:
copy/pasto: 'memory' is supposed to be 'memoryperjob'

Also: zero is a bad default for `$worker->{'hardware'}->{'jobs'}`, as this basically disables the check. Make this 1 instead.

Reply:
right, fixed.

Do you have a better idea for the name btw? it is IMHO not so readable with three words connected, but underscore would break the scheme...